CrossRoadController.cs

```diff
@@ -0,0 +1,374 @@
+        {
+            CrossRoadController crossRoadController = new();
+            do
+            {
+                time = TimeOfDay.TimeNow();
+                if (time >= TimeOfDay.DayModeStart() && time <= TimeOfDay.DayModeEnd())
+                {
+                    crossRoadController.CrossRoadControllerEvent += CrossRoadControllerEventArgs.ShowCrossRoadDayModeState;
+                    do
+                    {
+                        for (int i = 0; i < DayModeStatesCollection["States"].Count; i++)
+                        {
+                            time += cycleTime;
+                            cycleTime = 0;
+                            State = DayModeStatesCollection["States"][i];
+                            Main = DayModeStatesCollection["Main"][i];
+                            Secondary = DayModeStatesCollection["Secondary"][i];
+                            Tram = DayModeStatesCollection["Tram"][i];
+                            Pedestrian = DayModeStatesCollection["Pedestrian"][i];
+                            if (Main == "GREEN" || Secondary == "GREEN" || Pedestrian == "GREEN")
+                            {
+                                cycleTime += mainTimer;
+                                crossRoadController.CrossRoadControllerEvent?.Invoke(this, new CrossRoadControllerEventArgs(State, Main, Secondary, Tram, Pedestrian));
+                                await Task.Delay(mainTimer);
+                            }
+                            else if (Main == "REDYELLOW" || Secondary == "REDYELLOW" || Main == "YELLOW" || Secondary == "YELLOW")
+                            {
+                                cycleTime += yellowTimer;
+                                crossRoadController.CrossRoadControllerEvent?.Invoke(this, new CrossRoadControllerEventArgs(State, Main, Secondary, Tram, Pedestrian));
+                                await Task.Delay(yellowTimer);
+                            }
+                            else if (Main == "BLINKGREEN" || Secondary == "BLINKGREEN" || Pedestrian == "BLINKGREEN")
+                            {
+                                cycleTime += blinkTimer;
+                                crossRoadController.CrossRoadControllerEvent?.Invoke(this, new CrossRoadControllerEventArgs(State, Main, Secondary, Tram, Pedestrian));
+                                await Task.Delay(blinkTimer);
+                                cycleTime += blinkTimer;
+                                crossRoadController.CrossRoadControllerEvent?.Invoke(this, new CrossRoadControllerEventArgs(State, Main, Secondary, Tram, Pedestrian));
+                                await Task.Delay(blinkTimer);
+                                cycleTime += blinkTimer;
+                                crossRoadController.CrossRoadControllerEvent?.Invoke(this, new CrossRoadControllerEventArgs(State, Main, Secondary, Tram, Pedestrian));
+                                await Task.Delay(blinkTimer);
```

Comment:
Blinking logic can be implemented as a TrafficLight class logic and not the controller logic. Some traffic lights can blick others cannot. It is not a responsibility of the controller to tell the traffic light how to blink.